src/components/data/toolbar/Navigation.js

```diff
@@ -343,37 +353,63 @@ function Navigation(props) {
 
     const preProcessData = (respData) => {
         if (respData) {
-            const respRoots = respData.roots;
-            const home = userProfile
-                ? respRoots.find((root) => root.label === userProfile.id)
-                : {
-                      label: t("home"),
-                      path: `${irodsHomePath}/${constants.ANONYMOUS_USER}`,
-                  };
-            home.icon = <HomeIcon fontSize="small" />;
```

Comment:
Before I settled on creating the new hook functions for the base paths, I started refactoring this code, but I left some of this refactoring in this PR, since it didn't seem prudent to directly modify and use the objects returned by the service.

So now this function creates new objects with only the values this component needs from the service.

I also didn't see the point in constructing this anonymous user's home path here if `getFilesystemRoots` is already constructing it and returning it for logged out users.